core/renderers/block_rendering_rewrite/block_render_draw_highlight.js

```diff
@@ -50,176 +50,178 @@ goog.require('Blockly.blockRendering.Measurable');
+      this.steps_.push(this.notchPaths_.pathLeft);
+      this.steps_.push('h',  (this.RTL_ ? -0.5 : 0.5));
     } else if (elem.type === 'hat') {
-      this.highlightSteps_.push(
-          Blockly.blockRendering.constants.START_HAT.highlight(this.info_.RTL));
+      this.steps_.push(
+          this.startPaths_.path(this.RTL_));
     } else if (elem.isSpacer()) {
-      this.highlightSteps_.push('h', elem.width - Blockly.blockRendering.constants.HIGHLIGHT_OFFSET);
+      this.steps_.push('h', elem.width - this.highlightOffset_);
     }
   }
 
-  this.highlightSteps_.push('H', row.width - Blockly.blockRendering.constants.HIGHLIGHT_OFFSET);
+  this.steps_.push('H', row.width - this.highlightOffset_);
 };
 
 Blockly.blockRendering.Highlighter.prototype.drawValueInput = function(row) {
-  if (this.info_.RTL) {
-    var aboveTabHeight = -Blockly.blockRendering.constants.HIGHLIGHT_OFFSET;
-    var belowTabHeight = row.height -
-        Blockly.blockRendering.constants.PUZZLE_TAB.height +
-        Blockly.blockRendering.constants.HIGHLIGHT_OFFSET;
-    // Edge above tab.
-    this.highlightSteps_.push('v', aboveTabHeight);
-    // Highlight around back of tab.
-    this.highlightSteps_.push(
-        Blockly.blockRendering.constants.PUZZLE_TAB_HIGHLIGHT.pathDown(this.info_.RTL));
-    // Edge below tab.
-    this.highlightSteps_.push('v', belowTabHeight);
+  var input = row.getLastInput();
+  var pathString = '';
+  if (this.RTL_) {
+    var aboveTabHeight = -this.highlightOffset_;
+    var belowTabHeight =
+        row.height - input.connectionHeight + this.highlightOffset_;
+
+    pathString =
+        Blockly.utils.svgPaths.lineOnAxis('v', aboveTabHeight) +
+        this.puzzleTabPaths_.pathDown(this.RTL_) +
+        Blockly.utils.svgPaths.lineOnAxis('v', belowTabHeight);
   } else {
-    this.highlightSteps_.push(Blockly.utils.svgPaths.moveTo(row.width, row.yPos));
-    this.highlightSteps_.push(
-        Blockly.blockRendering.constants.PUZZLE_TAB_HIGHLIGHT.pathDown(this.info_.RTL));
+    pathString =
+        Blockly.utils.svgPaths.moveTo(row.width, row.yPos) +
+        this.puzzleTabPaths_.pathDown(this.RTL_);
   }
+
+  this.steps_.push(pathString);
 };
 
 Blockly.blockRendering.Highlighter.prototype.drawStatementInput = function(row) {
-  var x = row.statementEdge;
-  if (this.info_.RTL) {
-    this.highlightSteps_.push('M',
-        (x + Blockly.blockRendering.constants.DISTANCE_45_OUTSIDE) +
-        ',' + (row.yPos + Blockly.blockRendering.constants.DISTANCE_45_OUTSIDE));
-    this.highlightSteps_.push(
-        Blockly.blockRendering.constants.INNER_TOP_LEFT_CORNER_HIGHLIGHT_RTL);
-    this.highlightSteps_.push('v',
-        row.height - 2 * Blockly.blockRendering.constants.CORNER_RADIUS);
-    this.highlightSteps_.push(
-        Blockly.blockRendering.constants.INNER_BOTTOM_LEFT_CORNER_HIGHLIGHT_RTL);
+  if (this.RTL_) {
+    var innerHeight = row.height - (2 * this.insideCornerPaths_.height);
+    var steps =
```

Comment:
There's some inconsistent naming and structure between drawValueInput and drawStatementInput.
- One is a pathString and one is steps.
- One pushes to steps_ at the end, one pushes in each condition branch